crates/sui-graphql-rpc/src/extensions/timeout.rs

```diff
@@ -42,6 +44,13 @@ impl Extension for TimeoutExt {
         next: NextParseQuery<'_>,
     ) -> ServerResult<ExecutableDocument> {
         let document = next.run(ctx, query, variables).await?;
+        let is_mutation = document
```

Comment:
another noob q: parse_query and validation work on the entire request document right? Wouldn't the logic below set any query to `is_mutation = True` as long as one of the top-level operations is a mutation?

Reply:
Great question which made me think again about this. At least in the IDE, you cannot run both a query request and a mutation at the same time, so for mutations you have to specify `mutation {executeTransactionBlock(args) {fields}}`, so this would return True, and for a `query {}` would return false because the `OperationType` is `Query`